test/unit/integrations/paytm/paytm_module_test.rb

```diff
@@ -0,0 +1,68 @@
+require 'test_helper'
+require 'openssl'
+require 'base64'
+require 'digest'
+require 'securerandom'
+
+class PaytmModuleTest < Test::Unit::TestCase
+  include OffsitePayments::Integrations
+
+  def setup
+    OffsitePayments.mode = :test
+    @merchant_id = 'WorldP64425807474247'
+    @secret_key = 'kbzk1DSbJiV_O3p5'
+    @industry_type_id = 'Retail'
+    @website = 'worldpressplg'
+  end
+
+  def test_service_url_method
+    OffsitePayments.mode = :test
+    assert_equal 'https://pguat.paytm.com/oltp-web/processTransaction', Paytm.service_url
+
+    OffsitePayments.mode = :production
+    assert_equal 'https://secure.paytm.in/oltp-web/processTransaction', Paytm.service_url
+  ensure
+    OffsitePayments.mode = :test
+  end
+
+  def test_return_method
+    assert_instance_of Paytm::Return, Paytm.return('name=foo', {})
+  end
+
+  def test_notification_method
+    assert_instance_of Paytm::Notification, Paytm.notification('name=foo', {})
+  end
+
+  def test_checksum_method
+    paytm_load = { 'MID' => @merchant_id, 'ORDER_ID' => '100PT012', 'CUST_ID' => 'test@example.com', 'TXN_AMOUNT' => '10', 'CHANNEL_ID' => 'WEB', 'INDUSTRY_TYPE_ID' => @industry_type_id, 'WEBSITE' => @website, 'EMAIL' => 'test@example.com', 'MOBILE_NO' => '9999999999' }
```

Comment:
Hi Gaurav,

Kindly update the status.

Regards,

On Fri, Aug 5, 2016 at 11:37 PM, Gaurav Chande <notifications@github.com>
wrote:

> In test/unit/integrations/paytm/paytm_module_test.rb
> <https://github.com/activemerchant/offsite_payments/pull/200#discussion_r73735413>
> :
>
> > +    OffsitePayments.mode = :production
> > +    assert_equal 'https://secure.paytm.in/oltp-web/processTransaction', Paytm.service_url
> > +  ensure
> > +    OffsitePayments.mode = :test
> > +  end
> > +
> > +  def test_return_method
> > +    assert_instance_of Paytm::Return, Paytm.return('name=foo', {})
> > +  end
> > +
> > +  def test_notification_method
> > +    assert_instance_of Paytm::Notification, Paytm.notification('name=foo', {})
> > +  end
> > +
> > +  def test_checksum_method
> > +    paytm_load = { 'MID' => @merchant_id, 'ORDER_ID' => '100PT012', 'CUST_ID' => 'test@example.com', 'TXN_AMOUNT' => '10', 'CHANNEL_ID' => 'WEB', 'INDUSTRY_TYPE_ID' => @industry_type_id, 'WEBSITE' => @website, 'EMAIL' => 'test@example.com', 'MOBILE_NO' => '9999999999' }
>
> Also here ^
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/activemerchant/offsite_payments/pull/200/files/8d3a3cb36308be2f9337128b5e622051c16abfcd#r73735413>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL1PqPlHChihtadHSQoEqr4o_STrUvIsks5qc3vUgaJpZM4IQnF4>
> .
>



-- 
*Mohit Aggarwal*
Senior Software Engineer
+91 8826036707
[image: Inline image 1]
www.Paytm.com

-- 


Disclaimer :-

This email and any files transmitted with it are confidential and 
intended solely for the use of the individual or entity to which they are 
addressed. If you are not the intended recipient, you should not 
disseminate, distribute or copy this e-mail. Please notify the sender 
immediately by e-mail and destroy all copies of this message and any 
attachments. Any views or opinions presented in this email are solely those 
of the author and do not necessarily represent those of the company. 

Warning: Although the company has taken reasonable precautions to ensure no 
viruses are present in this email, the company cannot accept responsibility 
for any loss or damage arising from the use of this email or attachments.